src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -894,27 +894,137 @@ void UserInfoPopup::updateUserData()
     this->ui_.ignoreHighlights->setEnabled(false);
 }
 
-void UserInfoPopup::loadAvatar(const QUrl &url)
+void UserInfoPopup::loadAvatar(const HelixUser& user)
 {
-    QNetworkRequest req(url);
-    static auto manager = new QNetworkAccessManager();
-    auto *reply = manager->get(req);
+    this->avatarUrl_ = user.profileImageUrl;
+    auto filename = this->getFilename(user.profileImageUrl);
+    auto loaded = false;
+    auto sevenTVEnabled = getSettings()->displaySevenTVAnimatedProfile;
 
-    QObject::connect(reply, &QNetworkReply::finished, this, [=] {
-        if (reply->error() == QNetworkReply::NoError)
-        {
-            const auto data = reply->readAll();
+    QFile cacheFile(filename);
+    if (cacheFile.exists())
+    {
+        cacheFile.open(QIODevice::ReadOnly);
+        QPixmap avatar{};
+        avatar.loadFromData(cacheFile.readAll());
+        this->ui_.avatarButton->setPixmap(avatar);
+        loaded = true;
+    }
+    if (!loaded)
+    {
+        QNetworkRequest req(user.profileImageUrl);
+        static auto manager = new QNetworkAccessManager();
+        auto *reply = manager->get(req);
 
-            // might want to cache the avatar image
-            QPixmap avatar;
-            avatar.loadFromData(data);
-            this->ui_.avatarButton->setPixmap(avatar);
-        }
-        else
+        QObject::connect(reply, &QNetworkReply::finished, this, [=] {
+            if (reply->error() == QNetworkReply::NoError)
+            {
+                auto data = reply->readAll();
+
```

Comment:
```suggestion
```

Reply:
it seems ci is bugged here, perhaps you did this when github was down, please make 1 new commit.

Reply:
Or perhaps, make it so contributors can commit to your pr.